src/common/Modal.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useState} from "react";
+import ReactDOM from 'react-dom';
+import Modal from 'react-modal';
+import "./Modal.css"
+
+const customStyles = {
+    content : {
+        top: '50%',
+        left: '50%',
+        right: 'auto',
+        bottom: 'auto',
+        transform: 'translate(-50%, -50%)',
+        width: '600px',
+        height: '400px',
+        border: 'solid 1px #707070',
+        backgroundColor: '#ffffff',
+        borderRadius: '18px',
+        boxShadow: '10px 10px 6px 0 rgba(0, 0, 0, 0.16)'
+}
+
+};
+
+
+interface Props {
+    title: String,
+    text: String,
+    onClick : ()=>void,
+    button: JSX.Element
+}
+
+Modal.setAppElement('body')
+
+
+const ModalComponent = (props: Props) => {
```

Comment:
The modal component should not handle open and close functionality. The modal only is responsible for rendering something (ideally another component). Then the component that uses this modal should handle its open and its close.